### DISCUSSION TITLE
Transfer learning model refinement to change ljspeech

### DISCUSSION THREAD

**Discussion by JRMeyer** [coqui-ai/TTS#163] [Archived]
<USER>>>> hugo<br/> <DATE> [March 18, 2020, 10:10pm]<br/> <br/> Greetings folks! *slash* <br/> I am new to AI and learning stuff online and trying to fiddle around<br/> with whats available in GitHub and Colab. Also doing the fast.ai free<br/> online course to get a better understanding.<br/> <br/> The above preamble was meant to explain why my question below may seem<br/> noobish.<br/> <br/> I am interested in eventually using a pre-trained model and refining it<br/> to learn a different voice which would be used for TTS. Its a purely<br/> personal project so I don't have any quality standards to adhere to.<br/> <br/> My question is how does one go about achieving both of the following<br/> through model refinement (assuming i'm using a pre-trained LJSpeech<br/> model, and have adequate speech samples of the other voices)<br/> <br/> 1. Changing the voice to, using someone famous as an example, Dennis<br/> Quaid *slash* <br/> and simultaneously<br/> 2. Changing the delivery to, again using someone famous as an example,<br/> Morgan Freeman.<br/> <br/> Therefore, the output would be a refined voice TTS voice that sounds<br/> like Dennis but speaks with the cadence and delivery of Morgan.<br/> <br/> Is it actually possible to achieve that through transfer learning?<br/> <br/> Or, is this only possible to do by training a model from scratch on<br/> Morgan and then transfer learn to Dennis' voice?<br/> <br/> thanks!!<br/> <br/> [This is an archived TTS discussion thread from discourse.mozilla.org/t/transfer-learning-model-refinement-to-change-ljspeech]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> georroussos<br/> <DATE> [March 20, 2020, 8:02am]<br/> <br/> It works yes, I have done it. The dataset has to be relatively high<br/> quality and the duration of it should be approximately 6 hours and,<br/> after that, it took me 8 hours to fit.<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> hugo<br/> <DATE> [March 20, 2020, 5:32pm]<br/> <br/> Hi and thanks for the feedback George. That is very encouraging.<br/> <br/> I think I should be able to put together 6 hours each of both voices<br/> <br/> Can you also help me understand the sequence you applied to get good<br/> results. Is it the following: *slash* <br/> Step 1: Identify Training base: lJSpeech (i would use the Tacotron 260K<br/> model as the starting point) *slash* <br/> Step 2: Train/refine model against new voice 1 for about 8 hours to<br/> adapt prosody *slash* <br/> Step 3: Train/refine against new voice 2 for about 8 hours to get new<br/> voice/pitch<br/> <br/> Also, could you clarify how many steps you were able to complete in 8<br/> hours? That would give me an idea of what to plan for. I am still<br/> learning, but I understood that training might be different for a single<br/> GPU (or low batch size) training setup vs multiple GPUS or even another<br/> single GPU with higher ram or clockspeed<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> georroussos<br/> <DATE> [March 20, 2020, 6:31pm]<br/> <br/> Right, so the model I used was the one trained on forward attention and<br/> batch normalization. You get the model and you checkout the correct<br/> branch (most usually it's dev). You don't really need to pay attention<br/> to anything, really, besides making sure the dataset is as high of<br/> quality as possible. Mine was an internal one and it was a specific<br/> domain oriented, so it was pretty high quality, of approximately 6<br/> hours, male, however the transcriptions were not 100% correct. It took<br/> me approximately 8 hours until I got good results, but I don't know if<br/> it could've been better. The model was able to adapt both prosody and<br/> accent (which was mostly South American), together with voice.<br/> <br/> So you would need to get the model and the correct branch, along with<br/> the configuration and make sure you can continue training on the same<br/> model. Chances are you will get some errors because of the config file,<br/> in this case I am happy to help debug.<br/> <br/> I haven't worked with style modeling yet (prosody, accent etc), but I<br/> can relatively certainly tell you this approach is not able to get you a<br/> novel speaker, that is, for example, one speaker's voice with another's<br/> accent. That is probably because of the single-speaker limitation of the<br/> model. If you were to fine tune the model twice, you could just get the<br/> second voice's attributes. You can change the prosody either by using<br/> external embeddings, or global style tokens and providing a wav file as<br/> reference. Combining different voice attributes would probably take a<br/> multi-speaker model with embeddings support, which is what I am working<br/> on right now. I tried using embeddings on the single speaker model, but<br/> I only got as far as changing the prosody. If you're looking to do voice<br/> adaptation (source → target), you definitely need multispeaker and<br/> embeddings.<br/> <br/> In general, more GPUs do not imply higher convergence speed and most<br/> usually one GPU is enough.<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> hugo<br/> <DATE> [March 20, 2020, 8:16pm]<br/> <br/> Wow, thank for you the detailed explanation and sharing of your<br/> experience.<br/> <br/> Nice! So, 8 hours + a high quality dataset and you were able to change<br/> both prosody and voice to a new speaker. That is very interesting.<br/> <br/> I understand what you are saying about a different accent. I had thought<br/> of maintaining the same for both voices (probably US), so I'm probably<br/> okay there. I was not aware about the concept of external embeddings. I<br/> had seen the term 'speaker embedding' on the forum in a few posts, but<br/> did not make the connection. Thank you for bringing this up, I will read<br/> about it. So basically with this properly baked in one could potentially<br/> change the accent from US to US-Texan, or from US to UK-Scottish,<br/> without training from scratch on a huge dataset comprised of samples<br/> from the target accent.<br/> <br/> My main takeaway is that I should work on my training set sample before<br/> embarking upon any training to make sure it does not become the source<br/> of failure / time wasted in training.<br/> <br/> Thanks!<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> georroussos<br/> <DATE> [March 20, 2020, 9:53pm]<br/> <br/> Yes, the voice will change entirely to the new one<br/> It will be like training a new TTS, it only converges faster on the new<br/> voice because of the previous steps on LJSpeech.<br/> <br/> If you need help just give a man a shout. Best of luck!<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> srijan14<br/> <DATE> [June 23, 2020, 4:29pm]<br/> <br/> Do you happen to<br/> have any idea about transfer learning using the base LJSpeech model if I<br/> am training on a language other than english? *slash* <br/> Also is it possible to use the base LJSpeech model if i am using<br/> phonemic text for training.<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> erogol<br/> <DATE> [June 24, 2020, 10:09am]<br/> <br/> you can fine-tune LJSpeech model on a new language. Someone trained a<br/> new Turkish model using it with only 4 hours of speech.<br/> <br/> You can also train a model using phonemic text. You just need to provide<br/> the phoneme characters in config.json and disable native phoneme support<br/> in config.json and let it train.<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> Sadam1195<br/> <DATE> [June 24, 2020, 11:01am]<br/> <br/> I am assuming that speaker has to been same in both trained models<br/> right? Or does TTS support multi-speaker models as of now?<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> georroussos<br/> <DATE> [June 24, 2020, 2:47pm]<br/> <br/> Nope, the speaker does not need to be the same<br/> <br/> [Archived Post]<br/>
